services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/domain/Hanke.kt

```diff
@@ -56,5 +64,38 @@ data class Hanke(
     var polyHaitta: Haitta13? = null
     var tarinaHaitta: Haitta13? = null
 
+    /** Note: this can be null for two reasons; the field wasn't requested for, or there are no geometries for the hanke.
+     * See 'tilaOnGeometrioita' field.
+     */
     var geometriat: HankeGeometriat? = null
+
+    // --------------- State flags -------------------
+    var tilat: HankeTilat = HankeTilat()
+
+    fun updateStateFlags() {
+        updateStateFlagOnKaikkiPakollisetLuontiTiedot()
+        updateStateFlagTiedotLiikHaittaIndeksille()
+    }
+
+    fun updateStateFlagOnKaikkiPakollisetLuontiTiedot() {
+        // All mandatory fields have been given... (though their validity should be checked elsewhere)
+        //  and saveType is submit, not just draft?
+        tilat.onKaikkiPakollisetLuontiTiedot = !nimi.isNullOrBlank()
+                && !kuvaus.isNullOrBlank()
+                && (alkuPvm != null) && (loppuPvm != null)
+                && (vaihe != null) && (vaihe != Vaihe.SUUNNITTELU || suunnitteluVaihe != null)
+                && !tyomaaKatuosoite.isNullOrBlank()
+                && (kaistaHaitta != null) && (kaistaPituusHaitta != null)
+                && tilat.onGeometrioita == true
+                && saveType == SaveType.SUBMIT
```

Comment:
Why this rule?

Reply:
I thought we were already using that submit mode (in the end of the creation "wizard"), and thus, that prevents starting index analysis with something that is still in draft state, even if all the fields have been saved (with possibly placeholder values).